ts/features/idpay/details/utils/hooks.tsx

```diff
@@ -82,3 +91,47 @@ export const useInitiativeTimelineFetcher = (
     lastUpdate
   } as const;
 };
+
+export const useIdpayDiscountDetailsBottomSheet = () => {
```

Comment:
If this bottom-sheet is only used inside `InitiativeDetailsScreen.tsx`, what is the point of making it an utility function accessible from anywhere? Do you think this is a good design?

Reply:
honestly, I just don't like how long that screen was getting so I thought moving some parts outside would greatly improve readability for the main screen.
What do you think? yes adding another export to the codebase could be not so graceful, but so is adding another ~60 lines to an already huge file